File: C++/TowerOfHanoi.cpp
//  Tower of Hanoi

#include<iostream>

using namespace std;

void TowerOfHanoi(int n, char source, char destiny, char helper)
{
	if(n==0)
	{
		return ;
	}

	TowerOfHanoi(n-1,source,helper,destiny);
	cout << "Moving from " << source << " to " << destiny << endl ;
	TowerOfHanoi(n-1,helper,destiny,source);
}

int main()
{
	int n;
	cin >> n;
	TowerOfHanoi(n,'A','C','B');
	return 0;
}

File: Python/TowerOfHanoi.py
# Recursive Solution for Tower of Hanoi problem
def TowerOfHanoi(n, src, dst, tmp): 
    if n > 0:
        TowerOfHanoi(n - 1, src, tmp, dst) 
        print ("Move disk", n, "from rod", src, "to rod", dst )
        TowerOfHanoi(n - 1, tmp, dst, src) 

# Driver code
TowerOfHanoi(4, 'A', 'C', 'B')
